### ISSUE TITLE
Pressing back when a task is edited loses changes without prompt

### ISSUE THREAD

**Issue by h4xnoodle** [HabitRPG/habitica-android#1456]
Reproduction steps:
1. Start creating a task or edit a task
2. Make changes
3. Press 'hardware'/OS back button

Actual: user's changes are lost immediately
Expected: a prompt notifying user that their changes are not saved, and an option to save or cancel changes.

This is something I've experienced so much, and it's very annoying to lose my task list or changes if back is pressed. Typically the pattern is to ask a user to save or cancel their changes and at least warn them of dirty state.

I'm currently starting the changes for this issue.

**Comment by saraolson** [comment]
while that is definitely a pattern seen in more and more apps, all the task apps ive referenced have shown a send or save button, and differ on what happens when you tap back. google tasks auto saves, todoist shows the prompt suggested in this original issue, and anydo/daily just deletes the task. 

considering what current habitica users are accustomed to, i would still opt for the warning over auto save for now. we could then run analytics on how many times people tap back then tap save anyways vs discard and see for sure if tapping back to auto save is a pattern the majority of our users would expect.

**Comment by saraolson** [comment]
Strings/visual of how we'd like this modal to appear. 

<img width="358" alt="Screen Shot 2022-01-12 at 12 52 56 PM" src="https://user-images.githubusercontent.com/12453119/149195080-f77be593-c5d8-4c76-b678-e8cc0df58dbc.png">

**Comment by yarrouye** [comment]
Nice. The default should be "Cancel," though, not "Discard."

**Comment by saraolson** [comment]
We're testing making the current action of pressing back the main action. During testing, if more people tap cancel rather than discard we'll reevaluate the actions.
